.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,68 @@
+name: Integration Tests
+
+on:
+  pull_request:
+    paths:
+      - 'docs/example-library/**'
+      - 'integration/**'
+  workflow_dispatch:
+    inputs: {}
+
+jobs:
+  validate:
+    strategy:
+      matrix:
+        include:
+        - { integration_test: "meltano-basics", needs_postgres: false }
+        - { integration_test: "meltano-run", needs_postgres: true }
+      fail-fast: false
+
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out the repository
+      uses: actions/checkout@v3.0.2
+
+    - name: Install Poetry
+      env:
+        PIP_CONSTRAINT: .github/workflows/resources/constraints.txt
+      run: |
+        pipx install poetry
+        poetry --version
+
+    - name: Setup Python 3.8
+      uses: actions/setup-python@v4.0.0
+      with:
+        python-version: 3.8
+        architecture: x64
+        cache: 'poetry'
+
+    - name: Upgrade pip
+      env:
+        PIP_CONSTRAINT: .github/workflows/resources/constraints.txt
+      run: |
+        pip install pip
+        pip --version
+
+    - name: Install Dependencies
+      run: |
+        poetry env use "3.8"
+        poetry install
```

Comment:
In the long run would be good to replace `poetry install` with something like `pip install .` or `pipx install .`. Could expose issues related to the build process and/or dev dependencies being installed along with the prod dependencies.

In the long run, again, we could also sequence this behind the job @edgarrmondragon is working on which pre-builds the dist wheels and then install from the built wheels.

Reply:
@edgarrmondragon also found when working on the SDK that a migration from `tox` to `nox` would give an option to run in an environment that only uses the release libs. (Best in a separate issue.)